plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapManagerImpl.java

```diff
@@ -57,17 +57,22 @@
     private LdapContextFactory _ldapContextFactory;
 
     @Inject
-    private LdapUserManager _ldapUserManager;
+    private LdapConfiguration _ldapConfiguration;
+
+    @Inject LdapUserManagerFactory _ldapUserManagerFactory;
+
 
     public LdapManagerImpl() {
         super();
     }
 
-    public LdapManagerImpl(final LdapConfigurationDao ldapConfigurationDao, final LdapContextFactory ldapContextFactory, final LdapUserManager ldapUserManager) {
+    public LdapManagerImpl(final LdapConfigurationDao ldapConfigurationDao, final LdapContextFactory ldapContextFactory, final LdapUserManagerFactory ldapUserManagerFactory,
+                           final LdapConfiguration ldapConfiguration) {
```

Comment:
why inject and pass into constructor? is the ldap-provider-selector going to change on runtime?